core/src/test/scala/akka/persistence/cassandra/query/EventsByTagSpec.scala

```diff
@@ -851,3 +851,42 @@ class EventsByTagStrictBySeqNoEarlyFirstOffsetSpec
   }
 }
 
+class EventsByTagStrictBySeqNoManyInCurrentTimeBucketSpec
```

Comment:
I've tested this a view times, also locally in our test environment. The bug is not occurring anymore. So the solution is looking good! I still find it very difficult to determine the exact flow in this matter (probably also due to my lack of knowledge of the inner workings of the library), but the test is providing similar logs (when failing) as I had on my test env. The fix provided here does make sense imho.